src/query/storage/block.go

```diff
@@ -172,24 +234,39 @@ func (m *multiSeriesBlockSeriesIter) Next() bool {
 	return m.index < m.SeriesCount()
 }
 
-func (m *multiSeriesBlockSeriesIter) Current() (block.Series, error) {
+func (m *multiSeriesBlockSeriesIter) Current() (block.UnconsolidatedSeries, error) {
 	s := m.block.seriesList[m.index]
-	seriesLen := s.Values().Len()
-	values := make([]float64, m.block.StepCount())
-	seriesValues := s.Values()
+	values := make([]ts.Datapoints, m.block.StepCount())
+	seriesValues := s.Values().AlignToBounds(m.block.meta.Bounds)
+	seriesLen := len(seriesValues)
 	for i := 0; i < m.block.StepCount(); i++ {
 		if i < seriesLen {
-			values[i] = seriesValues.ValueAt(i)
+			values[i] = seriesValues[i]
 		} else {
-			values[i] = math.NaN()
+			values[i] = nil
```

Comment:
Seems a little dangerous to add nils here where we used to have NaNs, any chance that this could cause issues in scenarios where we assumed we were only getting NaNs and values?

Reply:
this is pointer to pointer now (ptr to datapoints). During consolidation, we would check for 0 datapoints, we should return NaN